File: 프로그래머스/1/92334. 신고 결과 받기/신고 결과 받기.py
def solution(id_list, report, k):
    r=[]
    # 신고 해시(딕셔너리) 리스트 신고당한 사람:[신고자 집합, 신고자 이메일 처리 횟수(이때는 키 값이 신고자가 된다.)]
    users={user:[set(),0] for user in id_list}
    # 신고 리스트를 돌면서
    for repo in report:
        # 신고 리스트 요소를 신고자와 신고당한 이용자로 분리 후
        # 신고 해시의 신고당한 유저 요소에 신고자 이름 추가
        id,re=repo.split(' ')
        if re not in users[re][0]:
            users[re][0].add(id)
    # 신고 딕셔너리 리스트를 돌면서
    for id, reported_ids in users.items():
        # 날 신고한 유저 수가 k값 이상이면
        if len(reported_ids[0])>=k:
            # 신고자 리스트 돌면서
            for r_id in reported_ids[0]:
                # 신고자 이메일 처리 횟수에 count
                users[r_id][1]+=1
    # 해시에서 값들만 뽑아낸 후 이메일 처리 횟수만 결과 리스트로 도출    
    r=[cnt for r_id,cnt in users.values()]
    return r

File: 프로그래머스/1/92334. 신고 결과 받기/신고 결과 받기.cpp
#include <string>
#include <vector>
#include <map>
#include <sstream>
#include <set>

using namespace std;

vector<int> solution(vector<string> id_list, vector<string> report, int k) {
    vector<int> r;
    string id,re;
    // 신고 해시(딕셔너리) 리스트 신고당한 사람:[신고자 집합, 신고자 이메일 처리 횟수(이때는 키 값이 신고자가 된다.)]
    map<string,pair<set<string>,int>> users;
    for(int i=0;i<id_list.size();i++) {
        users[id_list[i]]={{},0};
    }
    map<string,pair<set<string>,int>>::iterator itr;
    for(int i=0;i<report.size();i++) {
        // 신고 리스트 요소를 신고자와 신고당한 이용자로 분리
        istringstream iss(report[i]);
        iss>>id>>re;
        if(users[re].first.find(id)==users[re].first.end()) {
            users[re].first.insert(id);
        }
    }
    for(itr=users.begin();itr!=users.end();itr++) {
        if(itr->second.first.size()>=k) {
            for(auto it:itr->second.first) {
                users[it].second+=1;
            }
        }
    }
    for(int i=0;i<id_list.size();i++) {
        r.push_back(users[id_list[i]].second);
    }
    return r;
}